R/annotation-data.R

```diff
@@ -1,7 +1,7 @@
 #' Prepare promoter annotation data for the user specified txdb object
```

Comment:
`preparePromoterAnnotation` now takes in the file path of either a GTF/GFF or TxDb, detecting the file type, and building or loading the txdb accordingly.

Reply:
Allow for both file path `file` and `txdb` input. Check which argument is supplied with `missing` and generate txdb for downstream code.